go/fn/object_test.go

```diff
@@ -213,6 +213,49 @@ metadata:
 	}
 }
 
+func TestSetSlice(t *testing.T) {
+	var original = []byte(`apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: read-pods
+  namespace: default
+subjects:
+- kind: User
+  apiGroup: testing.group
+`)
+
+	var input = []byte(`apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: read-pods
+  namespace: default
```

Comment:
super super nit: usually I prefer using different name/namespace for each object. For two reasons
1. It better shows the purpose of the test. Update A fields to B fields. 
2. It reduces the possibility that the function does an entire replacement (not quite for your case since you call the `GetSlice`). But it is still a little bit confusing at the first glance.